src/Common/ZooKeeper/ZooKeeperLoadBalancer.cpp

```diff
@@ -0,0 +1,622 @@
+#include <algorithm>
+#include <deque>
+#include <memory>
+#include <mutex>
+
+#include <base/types.h>
+#include <base/sort.h>
+#include <base/getFQDNOrHostName.h>
+#include "Common/Logger.h"
+#include "Common/Priority.h"
+#include "Common/ZooKeeper/IKeeper.h"
+#include <Common/logger_useful.h>
+#include <Common/ZooKeeper/ZooKeeperLoadBalancer.h>
+#include <Common/ZooKeeper/ZooKeeperImpl.h>
+#include <Common/DNSResolver.h>
+#include <Common/isLocalAddress.h>
+
+#include <Poco/Logger.h>
+#include <Poco/Net/NetException.h>
+#include <Poco/String.h>
+
+
+enum Status
+{
+    UNDEF = 0,
+    ONLINE,
+    OFFLINE,
+};
```

Comment:
Do we really need 3 statuses? Once we get a status for a node, it immediately becomes outdated: an online node may suddenly go down, and an offline node may go up quickly. So it feels like the only status is UNDEF, we never know for sure. 

There are 2 main usecases:
 - We are connected to a suboptimal node. We can just try connecting to the optimal one, and switch if the attempt was successful; keep the suboptimal connection otherwise
 - Connection is lost and we need a new one. We can just try connecting to another node because only 1 of 3 nodes can be unavailable at the same time 

In both cases we can just connect optimistically and don't care about cached statuses